### PR TITLE
Existence and Membership filters for streams

### LINKED ISSUE #14: Support more filters
image
Currently Ax only supports "=" and "!=" would also support:

* Exists
* Does not exist
* Is one of
* Is not one of

via @romanlevin

### PR DESCRIPTION
**Pull request by romanlevin**
Fixes #14

### CODE REVIEW

File: cmd/ax/query.go [hunk 1]
```diff
@@ -24,6 +24,10 @@ func addQueryFlags(cmd *kingpin.CmdClause) *common.QuerySelectors {
 	cmd.Flag("after", "Results from after").StringVar(&flags.After)
 	cmd.Flag("select", "Fields to select").Short('s').HintAction(selectHintAction).StringsVar(&flags.Select)
 	cmd.Flag("where", "Add a filter").Short('w').HintAction(whereHintAction).StringsVar(&flags.Where)
+	cmd.Flag("one-of", "Add a membership filter (FIELD_NAME:FIELD_VALUE)").HintAction(oneOfHintAction).StringsVar(&flags.OneOf)
```
Comment (zefhemel) [inline cmd/ax/query.go]: Still the stylistic question, should this not be `where-one-of` to make it more "sentency"

File: cmd/ax/query.go [hunk 2]
```diff
@@ -80,6 +91,74 @@ func buildFilters(wheres []string) []common.QueryFilter {
 	return filters
 }
 
+var membershipFilterRegex = regexp.MustCompile(`([^!=<>]+)\s*:\s*(.*)`)
+
+func buildMembershipFilters(oneOfs []string, notOneOfs []string) []common.MembershipFilter {
+	// Build a nested map of field names and their membership constraints
+	fields := make(map[string]map[bool][]string)
```
Comment (zefhemel) [inline cmd/ax/query.go]: Wouldn't a `map[string]struct{ oneOf bool, value string}` be nicer? This is what this encodes right?

File: cmd/ax/query.go [hunk 3]
```diff
@@ -80,6 +91,74 @@ func buildFilters(wheres []string) []common.QueryFilter {
...
+		if !ok {
+			fields[fieldName] = map[bool][]string{
+				true:  {value},
+				false: {},
+			}
+		} else {
+			field[true] = append(fields[fieldName][true], value)
+		}
+	}
+	for _, notOneOfClause := range notOneOfs {
```
Comment (zefhemel) [inline cmd/ax/query.go]: This really feels like the same code twice, can this not be factored out somehow?

File: cmd/ax/query.go (hunk 2 above)
Comment (romanlevin) [reply cmd/ax/query.go]: Maybe, but this way we can just drop in the lists we generate.

File: cmd/ax/query.go (hunk 3 above)
Comment (romanlevin) [reply cmd/ax/query.go]: Extracted this into another function.

**Comment by romanlevin** [comment]
@zefhemel All the comments addressed, methinks.

**Pull request merged**